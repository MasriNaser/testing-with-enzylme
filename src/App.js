import React, { Component } from 'react';
import './App.css';

const Test = () => <div>Testing </div>;
class App extends Component {
  state={
    on:false,
    input: ''
  }
  render() {
    return (
      <div className='App'>
        <header className='App-header'>
          <h1>
            Edit <code>src/App.js</code> and save to reload.
          </h1>
          <a
            className='App-link'
            href='https://reactjs.org'
            target='_blank'
            rel='noopener noreferrer'
          >
            Learn React
          </a>
        </header>
        <ul className='naser'>
          <li>1</li>
          <li>2</li>
          <li>3333</li>
        </ul>
        <Test />
        <p className='button-state'>{this.state.on? 'yes':'No!'}</p>
        <button onClick={()=>this.setState({on:true})}>click here!</button>
        <h2>{this.state.input}</h2>
        <input type="text" onChange={(e)=> this.setState({
          input: e.currentTarget.value
        })}/>
      </div>
    );
  }
}
export class Link extends Component {
  render() {
    return this.props.hide ? null : <a href={this.props.address}>Click</a>;
  }
}

export default App;
