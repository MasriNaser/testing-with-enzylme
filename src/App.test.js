import React from 'react';
import ReactDOM from 'react-dom';
import App, { Link } from './App';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { writer } from 'repl';
configure({ adapter: new Adapter() });
//expect(wrapper.find('.App-header).exists()).toBe(true) to find exact element which has this className
describe('App', () => {
  const wrapper = shallow(<App />);
  //console.log(wrapper.debug())
  it('should has p', () => {
    //expect(wrapper.find('.App-header').length).toBe(1)
    //expect(wrapper.find('ul').children().length).toBe(3)
    //expect(wrapper.find('ul').hasClass('naser')).toBe(true)
    expect(wrapper.find('h1').text()).toBe(
      'Edit src/App.js and save to reload.'
    );
  });
  it('snapshot', () => {
    const tree = shallow(<App />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  //simulate a button
  it('on button click', () => {
    const wrapper = shallow(<App />);
    const button = wrapper.find('button');
    expect(wrapper.find('.button-state').text()).toBe('No!');
    button.simulate('click');
    expect(wrapper.find('.button-state').text()).toBe('yes');
  });
  //on change
  it('on change', () => {
    const wrapper = shallow(<App />);
    const input = wrapper.find('input');
    expect(wrapper.find('h2').text()).toBe('');
    input.simulate('change', { currentTarget: { value: 'masri' } });
    expect(wrapper.find('h2').text()).toBe('masri');
  });
  //new state
  it('update className',()=>{
    const wrapper = shallow(<App/>)
    expect(wrapper.find('.blue').length).toBe(1)
    expect(wrapper.find('.red').length).toBe(0)
    wrapper.setState({mainColor: 'red'})
    expect(wrapper.find('.blue').length).toBe(0)
    expect(wrapper.find('.red').length).toBe(1)
  })
});

describe('render the new component', () => {
  it('link component', () => {
    const wrapper = shallow(<Link address='www.google.com' />);
    expect(wrapper.instance().props.address).toBe('www.google.com');
  });
  it('a tag renders href', () => {
    const wrapper = shallow(<Link address='www.google.com' />);
    expect(wrapper.props().href).toBe('www.google.com');
  });
  it('return null with true hide prop', () => {
    const wrapper = shallow(<Link hide={false} />);
    expect(wrapper.find('a').length).toBe(1);
    wrapper.setProps({ hide: true });
    expect(wrapper.get(0)).toBeNull();
  });
});
