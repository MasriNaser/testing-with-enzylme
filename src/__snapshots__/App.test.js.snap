// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </h1>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
  <ul
    className="naser"
  >
    <li>
      1
    </li>
    <li>
      2
    </li>
    <li>
      3333
    </li>
  </ul>
  <Test />
  <p
    className="button-state"
  >
    No!
  </p>
  <button
    onClick={[Function]}
  >
    click here!
  </button>
  <h2 />
  <input
    onChange={[Function]}
    type="text"
  />
</div>
`;
